### PR TITLE
Add: search and help commands

### PR DESCRIPTION
**Pull request by yunginnanet**
Let me know if you'd like me to handle the delete within search thing differently. I just wanted it to give users a grace period.

Did this out of necessity and can confirm that it works well.

### CODE REVIEW

**Comment by Sora233** [comment]
Hi @yunginnanet thanks for your work.
Do we need `search` command while there has been a `keys` command already?
Or we can just add the `delete` option to `keys` command?

File: cli/grammar.go [hunk 1]
```diff
@@ -103,6 +104,44 @@ func (k KeysGrammar) Run(ctx *kong.Context, tx *buntdb.Tx) error {
...
+	fmt.Fprintf(ctx.Stdout, "\n\n-------------------\n\nFound %d keys, written above. \nSleeping 10 seconds before we delete.", len(keys))
+	for n := 0; n != 10; n++ {
+		time.Sleep(1 * time.Second)
+		fmt.Fprintf(ctx.Stdout, ".")
+	}
+
+	for _, k := range keys {
+		_, err := tx.Delete(k)
+		if err == nil {
+			fmt.Fprintln(ctx.Stdout, "Deleted: %v", k)
```
Comment (Sora233) [inline cli/grammar.go:139]: formatting directive %v?

File: cli/grammar.go [hunk 2]
```diff
@@ -103,6 +104,44 @@ func (k KeysGrammar) Run(ctx *kong.Context, tx *buntdb.Tx) error {
...
+
+	if !s.Delete {
+		return nil
+	}
+
+	fmt.Fprintf(ctx.Stdout, "\n\n-------------------\n\nFound %d keys, written above. \nSleeping 10 seconds before we delete.", len(keys))
+	for n := 0; n != 10; n++ {
+		time.Sleep(1 * time.Second)
+		fmt.Fprintf(ctx.Stdout, ".")
+	}
```
Comment (Sora233) [inline cli/grammar.go:134]: maybe need a newline when begine to delete

File: cli/grammar.go (hunk 1 above)
Comment (yunginnanet) [reply cli/grammar.go:139]: > Hi @yunginnanet thanks for your work. Do we need `search` command while there has been a `keys` command already? Or we can just add the `delete` option to `keys` command?

Keys only checks for a pattern against *keys*, not the contents of values

File: cli/grammar.go [hunk 3]
```diff
@@ -103,6 +104,47 @@ func (k KeysGrammar) Run(ctx *kong.Context, tx *buntdb.Tx) error {
...
+
+	fmt.Fprintf(ctx.Stdout, "\n\n-------------------\n\nFound %d keys, written above. \nSleeping 10 seconds before we delete.\n", len(keys))
+	for n := 0; n != 10; n++ {
+		time.Sleep(1 * time.Second)
+		fmt.Fprintf(ctx.Stdout, ".")
+	}
+
+	for i, k := range keys {
+		if i == 1 {
```
Comment (Sora233) [inline cli/grammar.go:137]: index start from 0

File: cli/grammar.go (hunk 1 above)
Comment (Sora233) [reply cli/grammar.go]: I see

File: cli/grammar.go (hunk 3 above)
Comment (yunginnanet) [reply cli/grammar.go:137]: ._.

yeah, I know. Sorry. One moment.

Comment (yunginnanet) [reply cli/grammar.go]:
```
./buntdb-cli ircd.db
ircd.db> search test --delete
account.channels spacedgod : #grandmas,#testingchannel,#crypto
account.channeltomodes ascii : {"#hacker":{"Modes":"o","JoinTime":1617768782678245684},"#security":{"Modes":"ov","JoinTime":1631590801019088379},"#tcpdirect":{"Modes":"v","JoinTime":1633680235671497123},"#test_":{"Modes":"o","JoinTime":1629172442518567297},"#vxug":{"Modes":"o","JoinTime":1633043084986268219}}
account.name test_ : test_
account.name test__ : test__
account.name testacctplzignore : testacctplzignore
channel.invitelist [redacted] : {"*!*@hgc.test":[redacted]
channel.name #testingchannel : #testingchannel

-------------------

Found 7 keys, written above.
Sleeping 10 seconds before we delete.
..........
Deleted: account.channels spacedgod
Deleted: account.channeltomodes ascii
Deleted: account.name test_
Deleted: account.name test__
Deleted: account.name testacctplzignore
Deleted: channel.invitelist #[redacted]
Deleted: channel.name #testingchannel
```

**Review by Sora233 — APPROVED** [review]
LGTM

**Pull request merged**